XCB setMouseCursor

fundies opened the pull request:
Adds setMouseCursor() function for linux as per issue #269. Please tell me what to put for doxygen comments and anything else to change.

```
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(800, 600), "My window");
    window.setMouseCursor(142); 

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }
        window.clear(sf::Color::Black);
        window.display();
    }

    return 0;
}
```

eXpl0it3r commented:
Can you please squash you two commits into one and use a proper commit message?

The discussion on this PR or the feature itself can be found on [the forum](http://en.sfml-dev.org/forums/index.php?topic=17299.0).

fundies commented:
No idea how to do either of those things.

eXpl0it3r commented:
http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

fundies commented:
Well, I only made things worse. I give up.

Mischa-Alff commented:
Your implementation uses the built-in cursor font instead of the themed font. You should look into how to use the themed font, or simply use `xcb-cursor`.

Also, what does `142` mean? You should have an enum or something with the common cursor types.

fundies commented:
I updated it to use xcb-cursor. Also I implemented setting cursor to image

xcb-cursor takes strings so new syntax is:
```
window.setMouseCursor("watch");
```
For images:
```
 sf::Image cursor;
	if (cursor.loadFromFile("pencil_1.png"))
		window.setMouseCursor(cursor.getSize().x, cursor.getSize().y, cursor.getPixelsPtr());
```

I dunno if i got all the tabs. I can't tell. Still no idea where / how I put / make enum so any guidence on that would be appricated.

select-this commented on src/SFML/Window/Unix/WindowImplX11.hpp:
```diff
@@ -228,19 +246,23 @@ class WindowImplX11 : public WindowImpl
+    bool                        m_isExternal;          ///< Tell whether the window has been created externally or by SFML
+    Atom                        m_atomClose;           ///< Atom used to identify the close event
+    int                         m_oldVideoMode;        ///< Video mode in use before we switch to fullscreen
+    bool						m_cursorVisible;       ///< Is the cursor visible
```
Still got some tabs here, but other than that they all seem to be gone.

fundies commented:
Windows and Linux now work. I'm having trouble figuring out os x...

binary1248 commented:
Superseded by #827.
